cmake/FindCybSDK.cmake

```diff
@@ -1,86 +1,34 @@
-# Copyright 2018 Istituto Italiano di Tecnologia (IIT)
-# @author Mohamed Babiker Mohamed Elobaid <mohamed.elobaid@iit.it>
+# Copyright 2020 Istituto Italiano di Tecnologia (IIT)
+# @author Kourosh Darvish <kourosh.darvish@iit.it>
 
-# Finds the Cyb SDK
+# Finds the Cyberith SDK
 #
-# This will define the following variables::
-#
-#   CybSDK_FOUND    - True if the system has the CybSDK 
-#   CybSDK_VERSION  - The version of the CybSDK  which was found
-#
-#
-
-##### Utility #####
-
-# Check Directory Macro
-macro(CHECK_DIR _DIR)
-  if(NOT EXISTS "${${_DIR}}")
-    message(WARNING "Directory \"${${_DIR}}\" not found.")
-    set(CybSDK_FOUND FALSE)
-    unset(_DIR)
-  endif()
-endmacro()
-
-# Check Files Macro
-macro(CHECK_FILES _FILES _DIR)
-  set(_MISSING_FILES)
-  foreach(_FILE ${${_FILES}})
-    if(NOT EXISTS "${_FILE}")
-      get_filename_component(_FILE ${_FILE} NAME)
-      set(_MISSING_FILES "${_MISSING_FILES}${_FILE}, ")
-    endif()
-  endforeach()
-  if(_MISSING_FILES)
-    message(WARNING "In directory \"${${_DIR}}\" not found files: ${_MISSING_FILES}")
-    set(CybSDK_FOUND FALSE)
-    unset(_FILES)
-  endif()
-endmacro()
-
-# Target Platform
-set(TARGET_PLATFORM)
-if(NOT CMAKE_CL_64)
-  set(TARGET_PLATFORM x86)
-else()
-  set(TARGET_PLATFORM x64)
+set(CybSDK_DIR $ENV{CybSDK_DIR} CACHE STRING "The directiry containing the Cyberith library")
+if(NOT DEFINED CybSDK_DIR)
+  message( FATAL_ERROR "variable {CybSDK_DIR} is not defined: ${CybSDK_DIR}" )
 endif()
 
-##### Find CybSDK #####
+option (BUILD_SHARED_LIBS "Please identify the library type SHARED or STATIC" FALSE)
```

Comment:
This line should go the main ``CMakeLists.txt`` if not already present.

Reply:
I am not sure, since if we add this line to the main CMakeLists, then we are setting this for all the libraries. In this case, it is only for this library.

Reply:
That's not an issue. Usually, you may want an entire project to be built statically, not just some libraries.

Reply:
yes, but what if we want we have some shared libraries only that we want to include it to the executable?
btw, by default this is False, so, all the libraries are built static.

Reply:
We quickly discussed F2F and a possible solution is to have a dedicated option to select which version to link. Here my suggestion:
```cmake
option (LINK_CYBSDK_SHARED "Please identify the if the CybSDK library should be SHARED or STATIC" ${BUILD_SHARED_LIBS})
```